saplings/circuits/src/components/forms/controls/index.js

```diff
@@ -0,0 +1,189 @@
+  children: PropTypes.oneOfType([
+    PropTypes.element,
+    PropTypes.arrayOf(PropTypes.element)
+  ])
+};
+
+InputWrapper.defaultProps = {
+  label: '',
+  error: '',
+  children: []
+};
+
+export const ListBoxSelect = ({
+  name,
+  label,
+  error,
+  value,
+  onChange,
+  options
+}) => {
+  let selectedOpt = options.find(opt => opt.value === value);
+  if (!selectedOpt && options.length > 0) {
+    [selectedOpt] = options;
+  } else if (!selectedOpt) {
+    selectedOpt = { value: '', content: '' };
+  }
+
+  return (
+    <InputWrapper label={label} error={error}>
+      <Dropdown
+        className="listbox-dropdown"
+        placeholderClassName="listbox-dropdown-placeholder"
+        controlClassName="listbox-dropdown-control"
+        menuClassName="listbox-dropdown-placeholder"
+        placeholder={selectedOpt.content}
+        options={options.map(opt => ({ value: opt.value, label: opt.content }))}
+        onChange={({ value: evtValue }) => {
+          const evt = {
+            preventDefault: () => null,
+            target: { name, value: evtValue }
+          };
+          onChange(evt);
+        }}
+      />
+    </InputWrapper>
+  );
+};
+
+ListBoxSelect.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.string,
+  label: PropTypes.string,
+  error: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.string.isRequired,
+      content: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
+        .isRequired
+    })
+  )
+};
+
+ListBoxSelect.defaultProps = {
+  value: '',
+  error: '',
+  label: '',
+  options: [{ value: '', content: '' }]
+};
+
+export const TextField = ({
+  name,
+  label,
+  value,
+  error,
+  children,
+  ...props
+}) => (
+  <InputWrapper label={label} error={error}>
+    {React.createElement('input', {
+      ...props,
+      type: 'text',
+      name,
+      value: value || ''
+    })}
+    {children}
+  </InputWrapper>
+);
+
+TextField.propTypes = {
+  name: PropTypes.string,
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  value: PropTypes.string,
+  error: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  children: PropTypes.oneOfType([
+    PropTypes.element,
+    PropTypes.arrayOf(PropTypes.element)
+  ])
+};
+
+TextField.defaultProps = {
+  name: '',
+  label: '',
+  value: '',
+  error: '',
+  children: []
+};
+
+export const InputRow = ({ className, children, ...props }) =>
+  React.createElement(
+    'div',
+    {
+      className: `input-row ${className}`,
+      ...props
+    },
+    children
+  );
+
+InputRow.propTypes = {
+  className: PropTypes.string,
+  children: PropTypes.oneOfType([
+    PropTypes.element,
+    PropTypes.arrayOf(PropTypes.element)
+  ])
+};
+
+InputRow.defaultProps = {
+  className: '',
+  children: []
+};
+
+export const Button = ({ label, type, ...props }) => {
+  // this circumvents a linting issue around type
```

Comment:
It does not like the use of the spread operator with JSX.  Seems like an arbitrary decision, since the use of the operator is extremely valuable for creating custom components that are still highly configurable.

Reply:
We disabled this lint in other recent projects

Reply:
I would prefer if we disabled the lint. I think a good approach for this would be to take the airbnb lint config (as it provides a good base), edit as we prefer, and then package it up. Then we can just pull in that commonly agreed upon config to every relevant project.

Reply:
We can do this in another PR